src/AppBundle/Controller/SocialController.php

```diff
@@ -819,55 +811,98 @@ public function textexportAction($decklist_id, Request $request)
             throw new NotFoundHttpException("Unable to find decklist.");
         }
 
-        $content = $this->renderView('AppBundle:Export:plain.txt.twig', [
-            "deck" => $decklist->getTextExport()
-        ]);
-        $content = str_replace("\n", "\r\n", $content);
+        $format = $request->query->get('format', 'text');
 
-        $response = new Response();
+        switch ($format) {
```

Comment:
Feedback is coming a bit late on this, since it got merged, but it's a minor thing anyway: this repetition between `SocialController` and `BuilderController` seems suspicious. The main difference seems to be one uses publicly available `Decklist` objects while the other deals directly with the private `Deck` object. Both derive from `ExportableDeck` which seems to have all the methods needed for the individual exports.

It may be possible to extract the export response logic to a separate, shared class. It's probably only beneficial to attempt to do so if you anticipate changing export more in the future (different export types, more sorting options, etc).

Reply:
absolutely. this can all be refactored out into either traits, or common base controller class, or a service or whatever. i do not want to tackle this at the moment, so i'm kicking that bucket down the road (hopefully not too far). thanks for the feedback!